### ISSUE TITLE
Some questions about rewards in the training process

### ISSUE THREAD

**Issue by CeruleanJ** [baturaysaglam/RIS-MISO-Deep-Reinforcement-Learning#10]
Greetings.Recently,I run a simulation with your code.I used the default values for all the parameters in the simulation without changing anything.I found that,at the end of the first episode,the reward can reach 11 or higher.
<img width="438" alt="eps 1" src="https://user-images.githubusercontent.com/131511005/233770122-32a629ac-3934-4397-8412-79cb9b716a22.png">
But,at the end of the second episode,the reward was stuck in 2 or less
<img width="469" alt="eps 2" src="https://user-images.githubusercontent.com/131511005/233770228-f8d3978c-df67-4164-8539-29dc7e4e465e.png">
What's more,the performance of subsequent episodes is much lower than that of the first episode.I don't know why it is like this. Shouldn't the performance of each episode increase as the number of training episode increases?Why is the performance best in the first episode？

**Comment by baturaysaglam** [comment]
Reinforcement learning algorithms are prone to sudden divergence (just a divergence out of nowhere) when they combine:

1. **Function approximation:** e.g., the use of neural networks)
2. **Off-policy learning:** e.g., sampling from the experience replay buffer)
3. **Bootstrapping:** what the conventional Q-learning based on

this issue is also known as the _deadly triad_ problem and has been widely studied in the RL literature. So, what you encounter is that the DDPG agent suffers from the deadly triad, and cannot improve anymore once it's diverged.

What you can do is simply consider a single episode. Determine the number of the step where the DDPG agent obtains good reward values. From now on, terminate the training when the DDPG agent reaches the specified number of training steps.   

Training it further is inclined to the deadly triad issue. However, the more advanced off-policy algorithms such as [Soft Actor-Critic](https://spinningup.openai.com/en/latest/algorithms/sac.html) are less prone to the deadly triad, which I empirically verified in my [paper](https://arxiv.org/abs/2211.09702). You may also want to take a look at [my paper's repo](https://github.com/baturaysaglam/RIS-MISO-PDA-Deep-Reinforcement-Learning), and there, you can find the implementation of SAC.

**Comment by CeruleanJ** [comment]
OK.Thanks for your help
